src/app/item-page/edit-item-page/item-status/item-status.component.ts

```diff
@@ -92,27 +147,66 @@ export class ItemStatusComponent implements OnInit {
       }
       operations.push(new ItemOperation('delete', this.getCurrentUrl(item) + '/delete', FeatureID.CanDelete, true));
       operations.push(new ItemOperation('move', this.getCurrentUrl(item) + '/move', FeatureID.CanMove, true));
-
       this.operations$.next(operations);
 
-      observableFrom(operations).pipe(
-        mergeMap((operation) => {
-          if (hasValue(operation.featureID)) {
-            return this.authorizationService.isAuthorized(operation.featureID, item.self).pipe(
-              distinctUntilChanged(),
-              map((authorized) => new ItemOperation(operation.operationKey, operation.operationUrl, operation.featureID, !authorized, authorized))
-            );
-          } else {
-            return [operation];
+      // Observable that reads identifiers and their status and, and config properties, and decides
+      // if we're allowed to show a Register DOI feature
+      let showRegister$: Observable<boolean> = combineLatest([this.identifiers$, registerConfigEnabled$]).pipe(
+        distinctUntilChanged(),
+        map(([identifiers, enabled]) => {
+          let no_doi = true;
+          let pending = false;
+          if (identifiers !== undefined && identifiers !== null) {
+            identifiers.forEach((identifier: Identifier) => {
+              if (hasValue(identifier) && identifier.identifierType === 'doi') {
+                // The item has some kind of DOI
+                no_doi = false;
+                if (identifier.identifierStatus === '10' || identifier.identifierStatus === '11'
```

Comment:
I've added a simple array to help translate the status into a string that is the same as the constant variable name used in the backend identifier code:
```
    public String[] statusText = {
        "UNKNOWN",                      // 0
        "TO_BE_REGISTERED",             // 1
        "TO_BE_RESERVED",               // 2
        "IS_REGISTERED",                // 3
        "IS_RESERVED",                  // 4
        "UPDATE_RESERVED",              // 5
        "UPDATE_REGISTERED",            // 6
        "UPDATE_BEFORE_REGISTRATION",   // 7
        "TO_BE_DELETED",                // 8
        "DELETED",                      // 9
        "PENDING",                      // 10
        "MINTED",                       // 11
    };
```

So at the time of constructing the HAL resource the string will populate here instead of the integer.